docker-examples/poetry-multistage/Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM python:3.11-slim as builder
+
+# --- Install Poetry ---
+ARG POETRY_VERSION=1.8
+
+ENV POETRY_HOME=/opt/poetry
+ENV POETRY_VIRTUALENVS_IN_PROJECT=1
+ENV POETRY_VIRTUALENVS_CREATE=1
+ENV PYTHONDONTWRITEBYTECODE=1
+ENV PYTHONUNBUFFERED=1
+# Tell Poetry where to place its cache and virtual environment
+ENV POETRY_CACHE_DIR=/opt/.cache
+
+RUN pip install "poetry==${POETRY_VERSION}"
+
+WORKDIR /app
+
+# --- Reproduce the environment ---
+# You can comment the following two lines if you prefer to manually install
+#   the dependencies from inside the container.
+COPY pyproject.toml poetry.lock /app/
+
+# Install the dependencies and clear the cache afterwards.
+#   This may save some MBs.
+RUN poetry install --no-root && rm -rf $POETRY_CACHE_DIR
```

Comment:
I wonder if `rm -rf $POETRY_CACHE_DIR` is necessary since this is the builder stage, that shouldn't (?) be present in the output image at all

Reply:
Having the `rm -rf $POETRY_CACHE_DIR` should keep the intermediate stage smaller.
With bigger dependencies (pytorch) the cache impact is notable.

Reply:
Why not using:
```
RUN --mount=type=cache,target=/opt/.cache \
    poetry install
```
This allows saving poetry package cache in local docker builder, which can be reused later. And it is not included into final image.